examples/hf/gpt2/pippy_gpt2.py

```diff
@@ -169,6 +153,15 @@ def run_master(_, args):
     parser.add_argument('--cuda', type=int, default=int(torch.cuda.is_available()))
     parser.add_argument('--record_mem_dumps', type=int, default=0, choices=[0, 1])
     parser.add_argument('--checkpoint', type=int, default=0, choices=[0, 1])
+    parser.add_argument('--rpc_timeout', type=int, default=1800)
+    parser.add_argument('--num_worker_threads', type=int, default=512)
+
+    parser.add_argument('--n_embd', type=int, default=None)
+    parser.add_argument('--n_layer', type=int, default=None)
+    parser.add_argument('--n_head', type=int, default=None)
+
+    parser.add_argument('--mode', type=str, default='small', choices=['small', 'large-cpu'])
+
```

Comment:
Please see my comment in #556 
If we can achieve that, then here we don't need to use `mode` to choose device. We would just need to pass a flag like `remote_materialize=[0,1]` to `PipelineDriver` by which `PipelineDriver` would ask each `PipeStageExecutor` to do so.